src/super_gradients/training/metrics/metric_utils.py

```diff
@@ -39,6 +39,40 @@ def calc_batch_prediction_detection_metrics_per_class(metrics, dataset_interface
     return results_tuple
 
 
+def calc_batch_prediction_detection_metrics_per_classV2(matching, dataset_interface, iou_thres, silent_mode,
+                                                      per_class_verbosity, class_names, test_loss):
+    images_counter = len(matching)
+    if len(matching):
+        matching_tensors = [torch.cat(x, 0) for x in list(zip(*matching))]
+        device = matching_tensors[0].device
+
+        ap, precision, recall, f1, unique_classes = compute_detection_metrics(*matching_tensors, device=device)
+        ap, precision, recall, f1 = ap.cpu().numpy(), precision.cpu().numpy(), recall.cpu().numpy(), f1.cpu().numpy()
+        _, unique_classes = unique_classes.cpu().numpy().astype(np.int64)
+
+        if iou_thres.is_range():
+            precision, recall, f1 = precision[:, 0], recall[:, 0], f1[:, 0]
+        mean_ap, mean_precision, mean_recall, mean_f1 = ap.mean(), precision.mean(), recall.mean(), f1.mean()
+    else:
+        targets_per_class = torch.zeros(1)
+
+    if not silent_mode:
```

Comment:
What are all of these prints ? We should only print what we usually do during training...

Reply:
this function was removed because outdated and never used: f5182088bd85175c86d345622c1242a924c989e7